scripts/adminrouter-agent-setup.ps1

```diff
@@ -0,0 +1,125 @@
+#======================================================================================
+# This script sets up the AdminRouter agent on a Windows node
+# This AdminRouter consists IIS, Application Request Routing and URL Rewrite with a 
+# set of a url rewrite rules
+#======================================================================================
+Param(
+    [Parameter(Mandatory=$true)]
+    [string]$AgentPrivateIP
+)
+
+Function Install-AdminModule{
+  Param(
+        [Parameter(Mandatory=$true)][string]$Name,
+        [Parameter(Mandatory=$true)][string]$MSISourceUri,
+        [Parameter(Mandatory=$true)][string]$DownloadDestFile,
+        [Parameter(Mandatory=$true)][string]$Logfile
+        )
+
+    Write-Output "Download and install $Name module"
+    Invoke-WebRequest -Uri $MSISourceUri -OutFile $DownloadDestFile
+    $arguments= ' /qn /l*v ' + $Logfile
+
+    Start-Process `
+         -file  $DownloadDestFile `
+         -arg $arguments `
+         -passthru | wait-process
+    if ($? -ne $true)
+    {
+        Write-Host "Could not install $Name module"
+    }
+
+}
+
+Function Install-URLRewriteAndARRModules{
+
+    $downloadDir = Join-Path $env:SystemDrive "dcos-download/"
+
+    # URL Rewrite
+    $UrlRewrite = "IIS URL Rewrite Module 2"
+    $UrlRewritelogfile = "rewrite.log"
+    [Uri]    $URLRewrite_Plugin_Uri  = 'http://download.microsoft.com/download/D/D/E/DDE57C26-C62C-4C59-A1BB-31D58B36ADA2/rewrite_amd64_en-US.msi'
+    [string] $URLRewrite_Plugin_File = $downloadDir + "rewrite_amd64_en-US.msi"
+
+    # Application Request Routing: AAR
+    $ARR = "Microsoft Application Request Routing 3.0"
+    $ARRlogfile = "apprequestrouting.log"
+    [Uri]    $ARR_Plugin_Uri = "https://download.microsoft.com/download/E/9/8/E9849D6A-020E-47E4-9FD0-A023E99B54EB/requestRouter_amd64.msi"
+    [string] $ARR_Plugin_File = $downloadDir + "requestRouter_amd64.msi"
+
+	New-Item -Path $downloadDir -ItemType directory -Force
+
+    Install-AdminModule -Name $UrlRewrite -MSISourceUri $URLRewrite_Plugin_Uri -DownloadDestFile $URLRewrite_Plugin_File -Logfile $ARRlogfile 
+    Install-AdminModule -Name $ARR -MSISourceUri $ARR_Plugin_Uri -DownloadDestFile $ARR_Plugin_File -Logfile $UrlRewritelogfile
+
+}
+
+function Install-IISAndTools{
+	$RequiredComponents = @("Web-Server","Web-Mgmt-Tools","Web-Request-Monitor", "Web-Http-Tracing")
+	foreach ($component in $RequiredComponents) {
+		$feature = get-windowsfeature | where { $_.Name -like $element }
+		if ($feature.InstallState -Eq "Installed"){
+		        Write-Host " 	$component was already installed"
```

Comment:
Can you use `Write-Output` instead of `Write-Host` everywhere ?
As far as I know, they are equivalent except a single behavior (that I know of).
 
`Write-Host` has a special behavior that it writes directly to the screen (and not standard output). As a consequence, you can't redirect any log messages written with `Write-Host` to a file, but you can with `Write-Output`.

To do a quick tests, simply try this in PowerShell:
```
PS C:\Users\ibalutoiu\Downloads> Write-Output "test" > test.txt
PS C:\Users\ibalutoiu\Downloads> cat .\test.txt
test

PS C:\Users\ibalutoiu\Downloads> Write-Host "test2" > test2.txt
test2
PS C:\Users\ibalutoiu\Downloads> cat .\test2.txt
```

In our case, we just do logging here, so `Write-Output` is more advisable.

Reply:
Thanks for the information. I will make the replacement changes,